src/pybind/mgr/dashboard/services/access_control.py

```diff
@@ -460,6 +461,18 @@ def create_user(self, username, password, name, email, enabled=True,
             self.users[username] = user
             return user
 
+    def save_attempt(self, username, attempt):
```

Comment:
Can you provide unit/integration tests to avoid regressions?

Reply:
I've included the tests. Maybe I can improve it, Could you take a look at it? @alfonsomthd